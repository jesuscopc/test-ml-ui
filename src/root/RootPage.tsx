import React from 'react';
import { useLocation } from 'react-router-dom';
import CardProduct from 'components/CardProduct/CardProduct';
import WithHeader from 'hocs/WithHeader/WithHeader';
import UseApiSearch, { TProduct } from 'hooks/UseApiSearch';

const RootPage = (): React.ReactElement => {
  const query = new URLSearchParams(useLocation().search);
  const searchPattern = query.get('search') || '';
  const { data } = UseApiSearch(searchPattern);
  return (
    <div className="container">
      {data.map((item: TProduct) => (
        <CardProduct key={item.title} {...item} />
      ))}
    </div>
  );
};

export default WithHeader(RootPage);
