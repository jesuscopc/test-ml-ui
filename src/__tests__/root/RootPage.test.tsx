import React from 'react';
import { shallow, mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import RootPage from 'root/RootPage';

describe('Suit test RootPage Page', () => {
  let wrapper = shallow(<RootPage />);
  beforeEach(() => {
    wrapper = shallow(<RootPage />);
  });
  test('should be render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should be render correctly', () => {
    const wrap = mount(
      <BrowserRouter>
        <RootPage />
      </BrowserRouter>
    );
    expect(wrap.find('.container')).toBeTruthy();
  });
});
