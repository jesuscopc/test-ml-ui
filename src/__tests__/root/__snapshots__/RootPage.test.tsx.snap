// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suit test RootPage Page should be render correctly 1`] = `
<Fragment>
  <div
    className="header"
  >
    <div
      className="container header__navBar"
    >
      <img
        alt="pagina principal"
        className="header__logo"
        src={Object {}}
        tabIndex={0}
      />
      <div
        className="header__searchBox"
      >
        <SearchBox />
      </div>
    </div>
  </div>
  <RootPage />
</Fragment>
`;
